macos/src/main/objectiveCpp/JNIDecorations.mm

```diff
@@ -47,87 +49,110 @@
+    [nsWindow release];
+JNF_COCOA_EXIT(env);
 }
 
 JNIEXPORT jdouble JNICALL
 Java_com_github_weisj_darklaf_platform_macos_JNIDecorationsMacOS_getTitleBarHeight(JNIEnv *env, jclass obj, jlong hwnd) {
-     NSWindow *nsWindow = OBJC(hwnd);
-     NSRect frame = NSMakeRect(0.0, 0.0, 100.0, 100.0);
-     NSUInteger windowStyle = nsWindow.styleMask & ~NSWindowStyleMaskFullSizeContentView;
-     NSRect rect = [NSWindow contentRectForFrameRect:frame styleMask:windowStyle];
-     return (jdouble)(frame.size.height - rect.size.height);
+JNF_COCOA_ENTER(env);
+    NSWindow *nsWindow = OBJC(hwnd);
+    NSRect frame = NSMakeRect(0.0, 0.0, 100.0, 100.0);
+    NSUInteger windowStyle = nsWindow.styleMask & ~NSWindowStyleMaskFullSizeContentView;
+    NSRect rect = [NSWindow contentRectForFrameRect:frame styleMask:windowStyle];
+    return (jdouble)(frame.size.height - rect.size.height);
+JNF_COCOA_EXIT(env);
+    return 0;
 }
 
 JNIEXPORT jboolean JNICALL
 Java_com_github_weisj_darklaf_platform_macos_JNIDecorationsMacOS_isFullscreen(JNIEnv *env, jclass obj, jlong hwnd) {
+JNF_COCOA_ENTER(env);
     NSWindow *nsWindow = OBJC(hwnd);
     return (jboolean)(([nsWindow styleMask] & NSWindowStyleMaskFullScreen) != 0);
+JNF_COCOA_EXIT(env);
+    return false;
 }
 
 JNIEXPORT jdouble JNICALL
 Java_com_github_weisj_darklaf_platform_macos_JNIDecorationsMacOS_getTitleFontSize(JNIEnv *env, jclass obj, jlong hwnd) {
+JNF_COCOA_ENTER(env);
     return (jdouble)[[NSFont titleBarFontOfSize:0] pointSize];
+JNF_COCOA_EXIT(env);
+    return 0;
 }
 
 JNIEXPORT void JNICALL
 Java_com_github_weisj_darklaf_platform_macos_JNIDecorationsMacOS_setTitleEnabled(JNIEnv *env, jclass obj, jlong hwnd,
                                                                                  jboolean enabled) {
+JNF_COCOA_ENTER(env);
     NSWindow *nsWindow = OBJC(hwnd);
-    dispatch_async(dispatch_get_main_queue(), ^{
+    [JNFRunLoop performOnMainThreadWaiting:YES withBlock:^{
         if (enabled) {
             nsWindow.titleVisibility = NSWindowTitleVisible;
         } else {
             nsWindow.titleVisibility = NSWindowTitleHidden;
         }
         [nsWindow contentView].needsDisplay = true;
-    });
+    }];
+JNF_COCOA_EXIT(env);
 }
 
 JNIEXPORT void JNICALL
 Java_com_github_weisj_darklaf_platform_macos_JNIDecorationsMacOS_setDarkTheme(JNIEnv *env, jclass obj, jlong hwnd,
                                                                               jboolean darkEnabled) {
-     NSWindow *nsWindow = OBJC(hwnd);
-     dispatch_async(dispatch_get_main_queue(), ^{
-          if (darkEnabled) {
-            nsWindow.appearance = [NSAppearance appearanceNamed:NSAppearanceNameDarkAqua];
-          } else {
-            nsWindow.appearance = [NSAppearance appearanceNamed:NSAppearanceNameAqua];
-          }
-          [nsWindow contentView].needsDisplay = true;
-     });
+JNF_COCOA_ENTER(env);
+    NSWindow *nsWindow = OBJC(hwnd);
+    if(!@available(macOS 10.14, *)) {
```

Comment:
```suggestion
    if(@available(macOS 10.14, *)) {
```

An alternative option is to check if  `NSAppearance appearanceNamed:...` produces `nil`.

Reply:
Technically it is not needed as the function only called if the version is at least `10.14`, but I figured it might be worth it to include the check nonetheless.